### PR TITLE
Prevent edge case that creates extra event loop

### PR DESCRIPTION
**Pull request by auxsvr**
Fixes https://github.com/scrapy/scrapy/issues/5831.

`asyncio.get_event_loop_policy().get_event_loop() is asyncio.get_running_loop()` is not always true, as mentioned in https://github.com/python/cpython/issues/96377#issuecomment-1260006538. The AsyncioSelectorReactor runs in the second thread and uses the already initialised event loop in the main thread to run the crawler, so a single event loop will be running code from two threads.

According to https://github.com/python/cpython/issues/100160, we might need to replace `asyncio.get_event_loop()` with `asyncio.get_running_loop()` if semantics change. Currently, some low level aspects of asyncio are in flux.

### CODE REVIEW

**Comment by Gallaecio** [comment]
I expect tests to break, we had to change this recently for a reason. Let’s see…

File: scrapy/utils/reactor.py [hunk 1]
```diff
@@ -90,7 +90,6 @@ def _get_asyncio_event_loop():
 
 def set_asyncio_event_loop(event_loop_path):
     """Sets and returns the event loop with specified import path."""
-    policy = get_asyncio_event_loop_policy()
```
Comment (wRAR) [inline scrapy/utils/reactor.py:93]: Not calling this function will break Windows. This is actually visible in the tests.

**Review by wRAR — CHANGES_REQUESTED** [review]

File: scrapy/utils/reactor.py (hunk 1 above)
Comment (auxsvr) [reply scrapy/utils/reactor.py:93]: It should work with acb62ea595a5cf8f4ec5fd97c17f03150de9d331.

The main problem at the moment is that testing these changes is tricky, as I don't use Windows. The only relevant and easy unit test I can think of is to check if `set_asyncio_event_loop() is asyncio.get_running_loop()`, but this tests some semantics more than anything else.

File: scrapy/utils/reactor.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def __call__(self):
         return self._func(*self._a, **self._kw)
 
 
-def get_asyncio_event_loop_policy():
+def set_asyncio_event_loop_policy():
```
Comment (Gallaecio) [inline scrapy/utils/reactor.py:57]: For backward compatibility reasons, we should keep the old function around, and log a deprecation warning when called.

**Review by Gallaecio — COMMENTED** [review]
Any chance you can add a test that only passes after your changes? To make sure we do not break what you are fixing here later on.

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Thanks!

**Pull request merged**